src/components/ui/dashboard/join-coaching-session.tsx

```diff
@@ -0,0 +1,153 @@
+  const TO_DATE = DateTime.now().plus({ month: 1 }).toISODate();
+
+  //@TODO: pass selected organization from organization array
+  const handleOrganizationSelection = (value: Id) => {
+    setOrganizationId(value);
+    if (value) {
+      fetchOrganization(value).then(([organization]) => {
+        organizationId = setOrganization(organization);
+      });
+    }
+  };
+
+  //@TODO: pass selected relationship from relationship array
+  const handleRelationshipSelection = (value: Id) => {
+    setRelationshipId(value);
+    if (value && organizationId) {
+      fetchCoachingRelationshipWithUserNames(organizationId, value).then(
+        (relationship) => {
+          setRelationshipId(relationship.id);
+          setCoachingRelationship(relationship);
+        }
+      );
+    }
+  };
+
+  const handleSessionSelection = (selectedSession: Id) => {
+    if (selectedSession && relationshipId) {
+      fetchCoachingSessions(relationshipId).then(([sessions]) => {
+        const theSession = sessions.find(
+          (session) => session.id === selectedSession
+        );
+        if (theSession) {
+          setCoachingSession(theSession);
+          setSessionId(theSession.id);
+        }
+      });
+    }
+  };
+
+  return (
+    <Card className="w-[300px]">
+      <CardHeader>
+        <CardTitle>Join a Coaching Session</CardTitle>
+      </CardHeader>
+      <CardContent className="grid gap-6">
+        <div className="grid gap-2">
+          <Label htmlFor="organization-selector">Organization</Label>
+
+          <DynamicApiSelect<Organization>
+            url="/organizations"
+            params={{ userId }}
+            onChange={handleOrganizationSelection}
+            placeholder="Select an organization"
+            getOptionLabel={(org) => org.name}
+            getOptionValue={(org) => org.id.toString()}
+            elementId="organization-selector"
+          />
+        </div>
+        {organizationId && (
+          <div className="grid gap-2">
+            <Label htmlFor="relationship-selector">Relationship</Label>
+
+            <DynamicApiSelect<CoachingRelationshipWithUserNames>
+              url={`/organizations/${organizationId}/coaching_relationships`}
+              params={{ organizationId }}
+              onChange={handleRelationshipSelection}
+              placeholder="Select coaching relationship"
+              getOptionLabel={(relationship) =>
+                `${relationship.coach_first_name} ${relationship.coach_last_name} -> ${relationship.coachee_first_name} ${relationship.coach_last_name}`
+              }
+              getOptionValue={(relationship) => relationship.id.toString()}
+              elementId="relationship-selector"
+            />
+          </div>
+        )}
+        {relationshipId && (
+          <div className="grid gap-2">
+            <Label htmlFor="session-selector">Coaching Session</Label>
+
+            <DynamicApiSelect<CoachingSession>
```

Comment:
For some reason I'm seeing duplicate past sessions and 1-2 of them are out of chronological order as well.

Reply:
I see that in the db as well. Not sure how that got added tbh.

Reply:
Oh you see it duplicated in the DB itself? In that case it's an error seeding data on the backend. I'll try changing it locally and see if this issue goes away.

Reply:
I found the culprit in the backend and fixed.